### PR TITLE
AVX512 version of mem_zero_detect()

### PR DESCRIPTION
**Pull request by Shark64**
Hi, as a test to try out AVX512 i rewrote mem_zero_detect() using those instructions. The mask registers turn out to be super useful to reduce code size and avoid branches 👍🏻 . This version should be faster than the AVX2 version while avoiding all the data dependent branches that were needed in the loop epilogue.
I don't have access to an AVX512 machine yet so i've run mem_zero_detect_test under the Intel software emulator to check if passes the tests. Please also test it on real hardware just to be sure.

### CODE REVIEW

**Comment by gbtucker** [comment]
Thanks @Shark64.  I'll take a look.

**Comment by Shark64** [comment]
Thanks. I'm not sure why the signoff check is failing, when i run check_format.sh on my locally it passes al the checks.
Is it something that i can fix with git?

**Comment by gbtucker** [comment]
> Thanks. I'm not sure why the signoff check is failing, when i run check_format.sh on my locally it passes al the checks. Is it something that i can fix with git?

Yes it's a little hard to follow also.  If you could squash all commits to one it would help both issues a lot.
```
git rebase origin/master
git rebase -i origin/master
<edit> with 
   pick
   s
   s
   ...
git commit -a -s --amend # fixup commit message and sign last
git push -f <your_upstream_repo> HEAD:<your_pr_br_name> # force push pr
```
After doing this it is more clear.  The .gitignore changes are probably not going to work. `*perf*` and `*test*` will include c files also.  Please drop the .gitignore changes.

**Comment by gbtucker** [comment]
Thanks @Shark64.  1. There seems to be an issue with github actions.  2. It just fails internal testing because Makefile.nmake is not up to date.
```
entering test: nmake_file_consistency
+ gmake -f Makefile.unx host_cpu=x86_64 test_nmake_file
Regenerating tst.nmake
--- Makefile.nmake	2021-12-20 22:11:06.000000000 -0500
+++ tst.nmake	2021-12-20 22:13:34.200899330 -0500
@@ -158,8 +158,9 @@
 	bin\igzip_gen_icf_map_lh1_04.obj \
 	bin\igzip_set_long_icf_fg_04.obj \
 	bin\igzip_set_long_icf_fg_06.obj \
-	bin\mem_zero_detect_avx.obj \
+	bin\mem_zero_detect_avx512.obj \
 	bin\mem_zero_detect_avx2.obj \
+	bin\mem_zero_detect_avx.obj \
 	bin\mem_zero_detect_sse.obj \
 	bin\mem_multibinary.obj
 
Potential nmake consistency issue
```
It can be updated with `make -f Makefile.unx Makefile.nmake` and `git commit -a --amend`

**Comment by Shark64** [comment]
@gbtucker thanks, i had totally missed that 😅 sorry. I'll update Makefile.name

**Comment by gbtucker** [comment]
Thanks.  Integrated.
